test/snippet/biocpp/part_3c_filtering_by_average_quality_of_paired_end.cpp

```diff
@@ -91,7 +92,7 @@ int main(int const argc, character_string argv[])
                        auto && [left_read, right_read] = pair;
                        return ranges::views::concat(std::views::single(std::move(left_read)), std::views::single(std::move(right_read)));
                    })
-                 | std::views::join;
+                 | seqan3::views::join_with;
```

Comment:
`std::views::join` gives an assert, not sure why

Reply:
do you know which one?

Reply:
```
part_3c_filtering_by_average_quality_of_paired_end_snippet: learning-resources/submodules/seqan3/submodules/range-v3/include/range/v3/view/concat.hpp:139: void ranges::concat_view<Rngs>::cursor<IsConst>::next_fun::operator()(ranges::indexed_element<I, N>) const [with I = seqan3::sequence_record<seqan3::type_list<std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::vector<seqan3::phred42, std::allocator<seqan3::phred42> > >, seqan3::fields<seqan3::field::seq, seqan3::field::id, seqan3::field::qual> >*; long unsigned int N = 1; bool IsConst = false; Rngs = {std::ranges::single_view<seqan3::sequence_record<seqan3::type_list<std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::vector<seqan3::phred42, std::allocator<seqan3::phred42> > >, seqan3::fields<seqan3::field::seq, seqan3::field::id, seqan3::field::qual> > >, std::ranges::single_view<seqan3::sequence_record<seqan3::type_list<std::vector<seqan3::dna5, std::allocator<seqan3::dna5> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::vector<seqan3::phred42, std::allocator<seqan3::phred42> > >, seqan3::fields<seqan3::field::seq, seqan3::field::id, seqan3::field::qual> > >}]: Assertion `it.get() != end(std::get<N>(pos->rng_->rngs_))' failed.
```

Reply:
gcc9 works, gcc10 gives a segfault, gcc11 has the assert

Reply:
Feels like this has something todo with it: https://github.com/gcc-mirror/gcc/blob/cad36f38576a6a781e3c62ab061c68f5b8dab13a/libstdc%2B%2B-v3/include/std/ranges#L2620
According to the debugger, rangev3 range doesn't to this check

Reply:
Hmm let's find later out why this does not work :)

This is also not so important, as this part will not be covered.